src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -186,7 +186,7 @@ private async Task<bool> AuthenticateAndAuthorizeAsync(HttpContext context, Func
 
         internal static bool RequiresAuthz(HttpRequest request, FunctionDescriptor descriptor)
         {
-            if (descriptor.Metadata.IsProxy())
+            if (descriptor.Metadata.IsProxy() || descriptor.IsWarmupFunction())
```

Comment:
@ankitkumarr Warmup functions are always anonymous? Users can't configure auth level in function.json?

Reply:
yes they are anonymous. and they are only being called by App Service FEs and Data roles. They are not external http triggers.